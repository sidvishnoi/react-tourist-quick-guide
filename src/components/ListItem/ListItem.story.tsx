import { text, withKnobs } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react';

import * as React from 'react';
import ListItem from '.';

const stories = storiesOf('Atoms/ListItem', module);
stories.addDecorator(withKnobs);

stories.add('in ordered list', () => {
  return (
    <ol>
      <ListItem>{text('text', 'hello')}</ListItem>
    </ol>
  );
});

stories.add('in unordered list', () => {
  return (
    <ul>
      <ListItem>{text('text', 'hello')}</ListItem>
    </ul>
  );
});
