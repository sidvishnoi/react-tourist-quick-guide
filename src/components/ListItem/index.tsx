import styled from 'styled-components';

const ListItem = styled.li`
  margin: 0.2rem 0;
`;

export default ListItem;
