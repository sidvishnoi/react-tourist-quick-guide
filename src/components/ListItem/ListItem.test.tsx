import React = require('react');
import * as renderer from 'react-test-renderer';

import ListItem from '.';

describe('Component - ListItem', () => {
  it('renders empty item', () => {
    const tree = renderer.create(<ListItem />);
    expect(tree).toMatchSnapshot();
  });

  it('renders plain text', () => {
    const tree = renderer.create(<ListItem>PASS</ListItem>);
    expect(tree).toMatchSnapshot();
  });
});
