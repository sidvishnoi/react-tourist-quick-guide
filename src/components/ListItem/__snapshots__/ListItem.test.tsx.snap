// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component - ListItem renders empty item 1`] = `
<li
  className="sc-bdVaJa DYeZc"
/>
`;

exports[`Component - ListItem renders plain text 1`] = `
<li
  className="sc-bdVaJa DYeZc"
>
  PASS
</li>
`;
